entity/contents/filter_contents.cc

```diff
@@ -34,6 +38,28 @@ std::shared_ptr<FilterContents> FilterContents::MakeBlend(
   return blend;
 }
 
+std::shared_ptr<FilterContents> FilterContents::MakeGaussianBlur1D(
+    InputVariant input_texture,
+    Scalar radius,
+    Point direction,
+    bool expand_border) {
+  auto blur = std::make_shared<GaussianBlur1DFilterContents>();
+  blur->SetInputTextures({input_texture});
+  blur->SetRadius(radius);
+  blur->SetDirection(direction);
+  blur->SetExpandBorder(expand_border);
+  return blur;
+}
+
+std::shared_ptr<FilterContents> FilterContents::MakeGaussianBlur(
+    InputVariant input_texture,
+    Scalar radius,
+    bool expand_border) {
+  auto x_blur =
+      MakeGaussianBlur1D(input_texture, radius, Point(1, 0), expand_border);
```

Comment:
I didn't see where these directional vectors were being transformed by the CTM.

Reply:
Ah sorry, what does CTM mean? The direction is normalized and passed to the shader via uniform buffer. The first pass is the X direction and the second pass is the Y direction, which is done on the output of the first blur.

Reply:
The radius and direction of the blur is relative to the transformation matrix. If you scale the scene by 2x the blur distance should double, if you scale it by non-uniform scale factors, the radius will be bigger in one axis than the other, if you skew the scene, then the blur distance along the skewed axes will be relative to the scale in that direction.

For example: https://fiddle.skia.org/c/156e36799e9c52b50947196d9575f9cf

Reply:
There is also a matrix on the gradient itself, but that isn't common to use. If it exists then the radii are further scaled/rotated/transformed/etc by that matrix in addition to the transformation matrix of the rendering operation.

Reply:
Skews and non-uniform scales and gradient transforms are rare, though, so for now it would suffice to make sure that the radius goes up with the transform (so that, for example, the Device Pixel Ratio affects them, but the above considerations are all important for completion.

Reply:
Oh, "current transformation matrix"? I thought that seemed familiar. 😄 

This is a good point. So right now, the blur actually "respects" transforms as you describe, but in a wrong way. The blend filter performs the blend _before_ the transform is applied to the _output_ image, and so there will be artifacting and too much/too little computation will occur depending on the transform.

I need to rewrite the filter class to do it the right way/allow filters to properly utilize transforms. Mind if I do that in a follow-up?

Reply:
Ah, yes, blur filters and transforms can be tricky and fun. For instance, if the blur radius is large then you can downsample and use a smaller radius without any noticeable loss of fidelity. If our sources have mipmaps then that might happen automatically. Then it becomes a series of tradeoffs for choosing how many samples to take compared to the resolution you are producing compared to the resolution of the source.